bokeh/io.py

```diff
@@ -310,7 +315,12 @@ def _show_with_state(obj, state, browser, new, notebook_handle=False):
     shown = False
 
     if state.notebook:
-        comms_handle = _show_notebook_with_state(obj, state, notebook_handle)
```

Comment:
I definitely think this should raise an informative `ValueError` if the notebook type is not `'jupyter'`